docs/data/data-grid/pagination/pagination.md

```diff
@@ -11,14 +11,20 @@ By default, each page contains 100 rows. The user can change the size of the pag
 
 ### Page size options
 
-You can configure the page size the user can choose from with the `pageSizeOptions` prop.
-
-It's possible to customize the options shown in the "Rows per page" select using the `pageSizeOptions` prop.
+You can customize the options shown in the "Rows per page" select using the `pageSizeOptions` prop.
 You should either provide an array of:
 
 - **numbers**, each number will be used for the option's label and value.
 
-- **objects**, the `value` and `label` keys will be used respectively for the value and label of the option (useful for language strings such as 'All').
```

Comment:
> useful for language strings such as 'All'

is false information.

```jsx
  <DataGrid pageSizeOptions={[{ value: -1, label: 'All' }]}>
  ```

Doesn't work

Reply:
Interesting. Should we support this use case anyways? I wonder if it'd be solved better by turning pagination off rather than supporting it with `pageSizeOptions`.